### ISSUE TITLE
Bug: [prefer-nullish-coalescing] This rule requires the `strictNullChecks` compiler option to be turned on to function correctly

### ISSUE THREAD

**Issue by frbuceta** [typescript-eslint/typescript-eslint#6280]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

None

### Repro Code

```typescript
https://github.com/loopbackio/loopback-next/pull/9110
```

### ESLint Config

```javascript
// Copyright IBM Corp. and LoopBack contributors 2019,2020. All Rights Reserved.
// Node module: @loopback/eslint-config
// This file is licensed under the MIT License.
// License text available at https://opensource.org/licenses/MIT

const fs = require('fs');

/**
 * Default ESLint configuration for LoopBack
 *
 * See https://eslint.org/docs/user-guide/configuring
 */
module.exports = {
  root: true,
  // Use the typescript-eslint parser
  parser: '@typescript-eslint/parser',
  // Enable eslint and typescript-eslint
  plugins: ['eslint-plugin', '@typescript-eslint', 'mocha'],
  env: {
    es6: true,
    node: true,
    mocha: true,
  },
  parserOptions: {
    sourceType: 'module',
    /*
     * The `project` setting is required for `@typescript-eslint/await-thenable`
     * but it causes significant performance overhead (1m13s vs 13s)
     * See https://github.com/typescript-eslint/typescript-eslint/issues/389
     */
    project: getProjectFile(),
    ecmaFeatures: {
      ecmaVersion: 2017,
      jsx: false,
    },
    noWatch: true,
  },
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    /**
     * Use `prettier` to override default formatting related rules
     * See https://github.com/prettier/eslint-config-prettier
     */
    'prettier',
  ],
  rules: {
    'prefer-const': 'error',
    'no-mixed-operators': 'off',
    'no-console': 'off',
    // 'no-undef': 'off',
    'no-inner-declarations': 'off',
    // TypeScript allows method overloading
    'no-dupe-class-members': 'off',
    'no-useless-escape': 'off',
    // TypeScript allows the same name for namespace and function
    'no-redeclare': 'off',

    /**
     * Rules imported from eslint-config-loopback
     */
    'mocha/handle-done-callback': 'error',
    'mocha/no-exclusive-tests': 'error',
    'mocha/no-identical-title': 'error',
    'mocha/no-nested-tests': 'error',
    'no-array-constructor': 'error',

    /**
     * TypeScript specific rules
     * See https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#supported-rules
     */
    '@typescript-eslint/array-type': 'off',
    '@typescript-eslint/indent': 'off',
    '@typescript-eslint/no-non-null-assertion': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/explicit-member-accessibility': 'off',
    '@typescript-eslint/no-var-requires': 'off',
    '@typescript-eslint/no-use-before-define': 'off',
    '@typescript-eslint/no-object-literal-type-assertion': 'off',
    '@typescript-eslint/no-parameter-properties': 'off',
    '@typescript-eslint/no-angle-bracket-type-assertion': 'off',
    '@typescript-eslint/prefer-interface': 'off',
    '@typescript-eslint/no-namespace': 'off',
    '@typescript-eslint/ban-types': 'off',
    '@typescript-eslint/no-triple-slash-reference': 'off',
    '@typescript-eslint/no-empty-interface': 'off',
    '@typescript-eslint/no-non-null-asserted-optional-chain': 'error',

    // Disable warning  Missing return type on function for now
    '@typescript-eslint/explicit-module-boundary-types': 'off',

    /**
     * The following rules are enforced to support legacy tslint configuration
     */
    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/ROADMAP.md
    // Rules mapped from `@loopback/tslint-config/tslint.common.json
    '@typescript-eslint/adjacent-overload-signatures': 'error', // tslint:adjacent-overload-signatures
    '@typescript-eslint/prefer-for-of': 'error', // tslint:prefer-for-of
    '@typescript-eslint/unified-signatures': 'error', // tslint:unified-signatures
    '@typescript-eslint/no-explicit-any': 'error', // tslint:no-any

    'no-unused-labels': 'error', // tslint:label-position
    'no-caller': 'error', // tslint:no-arg
    'no-new-wrappers': 'error', // tslint:no-construct
    // 'no-redeclare': 'error', // tslint:no-duplicate-variable

    'no-invalid-this': 'off',
    '@typescript-eslint/no-invalid-this': ['error'],
    '@typescript-eslint/no-misused-new': 'error',

    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md#how-to-use
    'no-shadow': 'off',
    '@typescript-eslint/no-shadow': 'error',

    'no-throw-literal': 'error', // tslint:no-string-throw

    '@typescript-eslint/no-unused-vars': [
      'error',
      {
        vars: 'all',
        args: 'none', // none - do not check arguments
        ignoreRestSiblings: false,
      },
    ], // tslint:no-unused-variable
    'no-unused-expressions': 'error', // tslint:no-unused-expression
    'no-var': 'error', // tslint:no-var-keyword
    eqeqeq: ['error', 'smart'], // tslint:triple-equals: [true, 'allow-null-check', 'allow-undefined-check'],

    // Rules mapped from `@loopback/tslint-config/tslint.build.json
    '@typescript-eslint/await-thenable': 'error', // tslint:await-promise: [true, 'PromiseLike', 'RequestPromise'],
    '@typescript-eslint/no-floating-promises': 'error',

    'no-void': 'error', // tslint:no-void-expression: [true, 'ignore-arrow-function-shorthand'],

    '@typescript-eslint/no-empty-function': 'off',
    '@typescript-eslint/consistent-type-assertions': 'off',
    '@typescript-eslint/no-misused-promises': 'error',

    '@typescript-eslint/prefer-optional-chain': 'error',
    '@typescript-eslint/prefer-nullish-coalescing': 'error',
    '@typescript-eslint/no-extra-non-null-assertion': 'error',

    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/return-await.md#how-to-use
    '@typescript-eslint/return-await': 'error',
    // note we must disable the base rule as it can report incorrect errors
    'no-return-await': 'off',

    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md
    camelcase: 'off',
    '@typescript-eslint/naming-convention': [
      'error',
      {
        selector: 'default',
        format: ['camelCase'],
      },

      {
        selector: 'variable',
        format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
      },

      {
        selector: 'variable',
        format: null,
        filter: '^_$',
      },

      // For mixin functions
      {
        selector: 'function',
        format: ['PascalCase'],
        filter: 'Mixin$',
      },

      {
        selector: 'parameter',
        format: ['camelCase'],
        leadingUnderscore: 'allow',
      },

      // For enum members
      {
        selector: 'enumMember',
        format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
        leadingUnderscore: 'allow',
      },

      // For properties
      {
        selector: 'property',
        format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
        leadingUnderscore: 'allow',
      },

      {
        selector: 'method',
        format: ['camelCase'],
        leadingUnderscore: 'allow',
      },

      // For static members
      {
        selector: 'memberLike',
        modifiers: ['static'],
        format: ['camelCase', 'UPPER_CASE'],
      },

      // For private members
      {
        selector: 'memberLike',
        modifiers: ['private'],
        format: ['camelCase'],
        leadingUnderscore: 'allow',
      },

      // For protected members
      {
        selector: 'memberLike',
        modifiers: ['protected'],
        format: ['camelCase'],
        leadingUnderscore: 'allow',
      },

      {
        selector: 'typeLike',
        format: ['PascalCase'],
      },

      {
        selector: 'objectLiteralProperty',
        format: null,
        // filter: '^([2-5]{1}[0-9]{2})$|[-/ ]',
        modifiers: ['requiresQuotes'],
      },
    ],
  },

  overrides: [
    {
      files: ['**/*.js'],
      rules: {
        '@typescript-eslint/prefer-optional-chain': 'off',
        '@typescript-eslint/prefer-nullish-coalescing': 'off',
      },
    },
  ],
};

/**
 * Detect tsconfig file
 */
function getProjectFile() {
  if (fs.existsSync('./tsconfig.build.json')) return './tsconfig.build.json';
  if (fs.existsSync('./tsconfig.json')) return './tsconfig.json';
  return undefined;
}
```

### tsconfig

```jsonc
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "compilerOptions": {
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "strict": true,

    // FIXME(bajtos) LB4 is not compatible with this setting yet
    "strictPropertyInitialization": false,

    // https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#use-unknown-catch-variables
    "useUnknownInCatchVariables": false,

    "incremental": true,

    "lib": ["es2020"],
    "module": "commonjs",
    "esModuleInterop": true,
    "moduleResolution": "node",
    "target": "es2018",
    "sourceMap": true,
    "declaration": true,
    "importHelpers": true
  }
}
```

### Expected Result

strict forces strictNullChecks to be true. So it shouldn't give an error. Even setting strictNullChecks doesn't work.

### Actual Result

https://github.com/loopbackio/loopback-next/actions/runs/3786273588/jobs/6437021406

**Comment by bradzacher** [comment]
Are you able to create an isolated repro for this? There's a lot of extraneous config and context here that makes it harder for us to investigate.

**Comment by frbuceta** [comment]
> Are you able to create an isolated repro for this? There's a lot of extraneous config and context here that makes it harder for us to investigate.

https://typescript-eslint.io/play#ts=4.9.3&sourceType=module&code=CYUwxgNghgTiAEYD2A7AzgF3hJBzXIwAkigKpogwpQC2IAXPJjAJYq4DcAUF8ulgFcKMNPAC88ANpd48AN7xqdRgCIA8gCMAnioA08KAUYBGAEzwAvrpnzFtBvBUAJEJSh6DR+AGZzVrgC63LyomNh4BMRkwuLwQpRoAHQAZmzAABTpAgCU4gB8cYlKCGKl4fiEJOSUxdncfGhIECCJOLjpbZFVwomGIHVAA&eslintrc=N4KABGBECmAeAu0B2ATAzpAXGA2uCUAAgDYD2pADgEYCGAxgNYD00axAlkvALR2lIAzdgHMWbTvABOdAHQArDABp8ESBWIBXYZ0wVJ0ePHbRJTfXwC2F5CmgpI+ALrLV6rZwzYca-YeOTIZ3xIUgA3E0l2W09cFTBQAgJIIWJWLFxIACpMpkyZeAwgxNVJDVSYhOLVC1I6AAsaJgbUVO4UfmheGmJiWkZ0kIEBByqAXzjxiEcQUaA&tsconfig=N4KABGBECmAeAu0B2ATAzpAXFAAgGwHsCAHAIwEMBjAawHpSBXASzxVsoKQDMmBzW+Gg7c+AOg4BbCZ1EArNJ0ggAvkA

Here he does not report the problem... It gives the example as valid. Which should not be happening.

**Comment by bradzacher** [comment]
Can you try creating an isolated repro repo to see if it reproduces with a minimal config? If it does then you can just link the repo here for us to look into.
